src/fragments/lib/auth/common/mfa/flows.mdx

```diff
@@ -0,0 +1,206 @@
+
+<b>AWS Cognito Limitation</b>
+
+The `fetchMFAPreference` API will only return enabled and preferred MFA methods, if the preferences were set using the `updateMFAPreference` API. If the preferences were not set using the `updateMFAPreference` API, the MFA preferences will return `nil` for enabled and preferred MFA methods. 
+
+Therefore it is recommended to use the `updateMFAPreference` API to set the MFA preferences for the user, if the user decides to set up any MFA methods. An example use case would be that when a user successfully sets up TOTP using the `setupTOTP` and `verifyTOTPSetup` API, `updateMFAPreference` should be invoked to set the MFA preferences for the user i.e. marking TOTP as enabled.
+
+</Callout>
+
+### Handling TOTP Setup during sign in
+
+A `signIn` call will return `continueSignInWithTOTPSetup` as a next step when the following set of conditions are met:
+
+- MFA is marked as **Required** in Cognito User Pool.
+- TOTP is marked as an enabled MFA method in Cognito user pool
+- User does not have any other MFA method set up.
+
+`continueSignInWithTOTPSetup` step signifies that the user must set up TOTP before they can sign in. The step returns an associated value of type `TOTPSetupDetails` which would be used for generating TOTP. `TOTPSetupDetails` provides a helper method called `getSetupURI` that can be used to generate a URI, which can be used by native password managers for TOTP association. For example. if the URI is used on Apple platforms, it will trigger the platform's native password manager to associate TOTP with the account. For more advanced use cases, `TOTPSetupDetails` also contains the `sharedSecret` that will be used to either generate a QR code or can be manually entered into an Authenticator app. 
+ 
+Once the authenticator app is set up, the user can generate a TOTP code and provide it to the library to complete the sign in process.
+
+import iosSignIn from "/src/fragments/lib/auth/ios/totp/sign_in.mdx";
+
+<Fragments fragments={{ios: iosSignIn}} />
+
+The TOTP code can be obtained from the user via a text field or any other means. Once the user provides the TOTP code, call `confirmSignIn` with the TOTP code as the `challengeResponse` parameter.
+
+import iosConfirmSignIn from "/src/fragments/lib/auth/ios/totp/confirm_sign_in.mdx";
+
+<Fragments fragments={{ios: iosConfirmSignIn}} />
+
+### Enabling TOTP after a user is signed in
+
+TOTP MFA can be set up after a user has signed in. This can be done when the following conditions are met:
+
+- MFA is marked as **Optional** or **Required** in Cognito User Pool.
+- TOTP is marked as an enabled MFA method in Cognito user pool
+
+TOTP setup can be achieved by calling `setUpTOTP` and `verifyTOTPSetup` APIs that are available in the `Auth` category.
+
+Invoke the `setUpTOTP` API to generate a `TOTPSetupDetails` object that contains the `sharedSecret` which will be used to either to generate a QR code or can be manually entered into an Authenticator app.
```

Comment:
I would more-or-less copy the spiel from above and mentioning `getSetupUri`. The inclusion of shared secret should be an afterthought in the discussion IMO.